distributed/utils.py

```diff
@@ -334,7 +334,16 @@ class SyncMethodMixin:
     @property
     def asynchronous(self):
         """Are we running in the event loop?"""
-        return in_async_call(self.loop, default=getattr(self, "_asynchronous", False))
+        return getattr(self, "_asynchronous", False) or in_async_call(self.loop)
+
+    @asynchronous.setter
```

Comment:
Given that we might re-use objects like `Client`s (potentially concurrently), it feels unsafe to me to allow us to switch between `async` and `sync behavior`. I think it might be safer to follow the suggestion in #7658

> My proposal would be, in the case of a default or current client to raise a RuntimeError in case the default/current doesn't correspond to the appropriate asnychronous setting.

What are your thoughts?

Reply:
Yep, painfully obvious now this was a silly assumption on my end. :)  https://github.com/dask/distributed/pull/7871#discussion_r1227120572